app/scripts/services/builds.js

```diff
@@ -2,152 +2,85 @@
-    BuildsService.prototype.cloneBuild = function(buildName, context, $scope) {
-      var req = {
-        kind: "BuildRequest",
-        apiVersion: "v1",
-        metadata: {
-          name: buildName
-        }
-      };
-      DataService.create("builds/clone", buildName, req, context).then(
-        function(build) { //success
-            $scope.alerts = $scope.alerts || {};
-            var logLink = $filter('buildLogURL')(build);
-            var alert = {
-              type: "success",
-              message: "Build " + buildName + " is being rebuilt as " + build.metadata.name + "."
-            };
-            if (logLink) {
-              alert.links = [{
-                href: logLink,
-                label: "View Log"
-              }];
-            }
-            $scope.alerts["rebuild"] = alert;
-        },
-        function(result) { //failure
-          $scope.alerts = $scope.alerts || {};
-          $scope.alerts["rebuild"] =
-            {
-              type: "error",
-              message: "An error occurred while rerunning the build.",
-              details: $filter('getErrorDetails')(result)
-            };
-        }
-      );
-    };
+    var annotation = $filter('annotation');
 
-    BuildsService.prototype.isPaused = function(buildConfig) {
+    var isPaused = function(buildConfig) {
       return $filter('annotation')(buildConfig, "openshift.io/build-config.paused") === 'true';
     };
 
-    BuildsService.prototype.canBuild = function(buildConfig) {
-      if (!buildConfig) {
-        return false;
-      }
-
-      if (buildConfig.metadata.deletionTimestamp) {
-        return false;
-      }
-
-      if (this.isPaused(buildConfig)) {
-        return false;
-      }
-
-      return true;
-    };
-
-    var annotation = $filter('annotation');
-    BuildsService.prototype.usesDeploymentConfigs = function(buildConfig) {
-      var uses = annotation(buildConfig, 'pipeline.alpha.openshift.io/uses');
-      if (!uses) {
-        return [];
-      }
+    return {
+      startBuild: function(buildConfigName, context) {
+        var req = {
+          kind: "BuildRequest",
+          apiVersion: "v1",
+          metadata: {
+            name: buildConfigName
+          }
+        };
+        return DataService.create("buildconfigs/instantiate", buildConfigName, req, context);
+      },
+      cancelBuild: function(build, buildConfigName, context) {
+        var canceledBuild = angular.copy(build);
+        canceledBuild.status.cancelled = true;
+        return DataService.update("builds", canceledBuild.metadata.name, canceledBuild, context);
+      },
+      cloneBuild: function(buildName, context) {
+        var req = {
+          kind: "BuildRequest",
+          apiVersion: "v1",
+          metadata: {
+            name: buildName
+          }
+        };
+        return DataService.create("builds/clone", buildName, req, context);
+      },
+      // Q: should this be in a buildConfig service?
```

Comment:
what are you asking here exactly?  The build config is what is being paused, not a particular build.  it means that builds can not be launched from the build config currently